sample-apps/SampleAppSpec.md

```diff
@@ -0,0 +1,233 @@
+
+#### How the Sample Apps should work
+
+Each sample app should emit the same Metrics, Logs and Traces. If a sample app cannot emit the mentioned, then that will noted in that sample app’s documentation.
+Each Sample app should should be able to be built as an image. The sample app must be able to accept configuration through a configuration file (YAML) and environment variables.
+
+#### Assumption on how the Sample Apps should emit Traces and Metrics
+
+**Traces** are not something that could or should be generated arbitrarily but instead should reflect a real life cycle or flow.
+In existing sample applications, Trace instrumentation is coupled with a **Traffic Generator** (https://github.com/aws-observability/aws-otel-test-framework/tree/terraform/traffic-generator) that makes http requests to the Sample app and emits traces based on that those requests. In another example, **ho11y** (https://gitlab.aws.dev/hausenbl/); sample apps can create their own traffic by invoking other sample apps (invocation pattern). 
+
+
+Sample apps in conformance to this document will also include the invocation pattern as an optional configuration. This will allow for more extensive testing and demos if configured.
+A possible example can be seen below. This invocation capability can allow testing to be as complex or simple as needed.
+
+![Alt text](./invocation.png?raw=true "Invocation")
+
+The two patterns of metrics that will be generated are **Request-based** and **Random**. Request based metrics will also assume an invocation has been made to the Sample App. Random can be arbitrarily generated for metrics that do not correlate to incoming requests. 
+
+### Documentation Requirements
+
+Each sample app MUST be coupled with a document that will have the following sections:
+
+* **Description** - Information about the language being used and a link to this document for context
+* **Non-conformance** - Documentation on missing features or missing extensions that don’t allow a given sample app to conform with this document
+* **Workarounds** - Examples of workarounds include the use of an unstable API or SDK to produce signals. This code `workaround` code may require changes in the future.
+* **README** - The readme that will go into each sample app’s code base detailing how to setup the sample app (this should be same across all sample apps); although redundant, each repository should still have it to avoid looking across sample apps for a readme. This readme should also include the Non-conformance section specific to the sample app it belongs to.
+
+### Functional Requirements
+
+Each sample app should be configured with manual instrumentation from OTel. Auto instrumentation can also be implemented.
+Metrics and traces should be OTLP and Logs should be in JSON format using any popular logging library for each respective language with the appropriate license. (e.g. Log4j for java, Logrus for Go)
```

Comment:
Yes it seems to be referring to the use of the OTLP grpc exporter: https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/otlpexporter/README.md.  It seems like Logs are in beta for this exporter so should we use it for logs as well so that all signals use the OTLP exporter?

Reply:
I think that would make sense

Reply:
Ok I have updated the Spec to reflect that all signals will use the OTLP exporter